dist/index.js

```diff
@@ -3836,46 +3829,59 @@ function run() {
                 'meta.source': 'gha-buildevents'
             });
             // create a first step to time installation of buildevents
-            yield buildevents.step(traceId, util.randomInt(Math.pow(2, 32)).toString(), buildStart.toString(), 'gha-buildevents_init');
+            const initStepComponents = ['gha-buildevents_init', util.getEnv('GITHUB_JOB'), core.getInput('matrix-key')];
+            yield buildevents.step(traceId, util.randomInt(Math.pow(2, 32)).toString(), buildStart.toString(), util.replaceSpaces(initStepComponents.filter(value => value).join('-')));
             core.info('Init done! buildevents is now available on the path.');
-            // set TRACE_ID to be used throughout the workflow
-            util.setEnv('TRACE_ID', traceId);
             // save buildStart to be used in the post section
             core.saveState('buildStart', buildStart.toString());
             core.saveState('isPost', 'true');
+            if (core.getInput('status') || core.getInput('job-status')) {
+                core.saveState('endTrace', 'true');
+            }
         }
         catch (error) {
             core.setFailed(error.message);
         }
     });
 }
 function runPost() {
-    var _a;
     return __awaiter(this, void 0, void 0, function* () {
         try {
             const postStart = util.getTimestamp();
-            const traceId = (_a = util.getEnv('TRACE_ID')) !== null && _a !== void 0 ? _a : '0';
-            const buildStart = core.getState('buildStart');
-            const jobStatus = core.getInput('job-status', { required: true });
-            const result = jobStatus.toUpperCase() == 'SUCCESS' ? 'success' : 'failure';
+            const traceId = buildTraceId();
+            // use trace-start if it's provided otherwise use the start time for current job
+            const traceStart = core.getInput('trace-start') ? core.getInput('trace-start') : core.getState('buildStart');
+            // if status is empty, grab legacy job-status value
+            const workflowStatus = core.getInput('status') ? core.getInput('status') : core.getInput('job-status');
+            const result = workflowStatus.toUpperCase() == 'SUCCESS' ? 'success' : 'failure';
             buildevents.addFields({
-                'job.status': jobStatus
+                'workflow.status': workflowStatus
```

Comment:
While jobStatus is deprecated, I think it makes sense to send both of these trace fields for the moment (as people may be relying on `job.status` in dashboards or queries as well)

```suggestion
                'job.status': workflowStatus,
                'workflow.status': workflowStatus
```

Reply:
good call, thanks!